Sdk/Speckle.Connectors.Utils/Operations/ReceiveOperation.cs

```diff
@@ -50,15 +53,49 @@ IServerTransportFactory serverTransportFactory
     using var transport = _serverTransportFactory.Create(account, receiveInfo.ProjectId);
     using (var _ = SpeckleActivityFactory.Start("Receive objects"))
     {
+      _progressDisplayManager.Begin();
       commitObject = await Speckle
         .Sdk.Api.Operations.Receive(
           version.referencedObject,
           transport,
           onProgressAction: dict =>
           {
+            if (!_progressDisplayManager.ShouldUpdate())
+            {
+              return;
+            }
             // NOTE: this looks weird for the user, as when deserialization kicks in, the progress bar will go down, and then start progressing again.
             // This is something we're happy to live with until we refactor the whole receive pipeline.
-            onOperationProgressed?.Invoke($"Downloading and deserializing", dict.Values.Average() / totalCount);
+            var args = dict.FirstOrDefault();
+            if (args is null)
+            {
+              return;
+            }
+            switch (args.ProgressEvent)
+            {
+              case ProgressEvent.DownloadBytes:
+                onOperationProgressed?.Invoke(
```

Comment:
`onOperationProgressed?.Invoke("Downloading", percentage, speedText);` that would help to not break debouncing